cc/toolchains/cc_toolchain_info.bzl

```diff
@@ -119,6 +119,12 @@ FeatureInfo = provider(
         "allowlist_include_directories": "(depset[DirectoryInfo]) Include directories implied by this feature that should be allowlisted in Bazel's include checker",
     },
 )
+
+FeatureImplyabilityInfo = provider(
```

Comment:
Any feature that returns `FeatureSetInfo` should be implyable, as they can be toggled by the toolchain via `known_features` and `enabled_features`.

I'd prefer you removed `FeatureImplyabilityInfo` and just made `cc_target_capability` not return a `FeatureSetInfo`, and added a seperate `target_capabilities` attribute to the toolchain configuration, as it's neither a known feature nor an enabled feature.

Reply:
Do we really want to add a dedicated attribute for something we've already decided is an anti-pattern though?

To fully expand this problem space(sans hidden behaviors that are undocumented):
* `supports_pic`: platform specific, controlled by a feature, has interactions with `--force_pic`.
* `supports_dynamic_linker`: platform specific, controlled by a feature.
* `supports_start_end_lib`: tool-dependent, controlled by a feature and `--start_end_lib`.
* `supports_interface_shared_libraries`: tool dependent, controlled by a feature and `--interface_shared_objects`.
* `supports_header_parsing`: tool dependent, controlled by an attribute on `cc_toolchain`, [marked as deprecated in its usage in code](https://github.com/bazelbuild/bazel/blob/1c47c9d412b29f7ba42c190e0b211a9438a81d59/src/main/java/com/google/devtools/build/lib/rules/cpp/CcCommon.java#L409-L411).
* `supports_param_files`: tool dependent, controlled by an attribute on `cc_toolchain`. 

There's a bunch of fragmentation here. I'm happy with `cc_tool_capability`, with the minor caveat that clearly some tool capabilities can't cleanly be expressed that way today.

It's starting to feel like we have other paths for these two oddball cases that are target-specific:
1. Make the fact that they're features purely an implementation detail, and just expose them as direct bool attributes of a `cc_toolchain` configuration a la `supports_header_parsing`. Users won't be able to express feature relationships with these any more, but I'm not seeing any reason that they should or any instances where Bazel relies on this being possible.
2. Maybe they should just be `bool_flag`s, which lets platforms control the behavior directly at the expense of losing any correlation with the toolchain configuration. This seems like the most "normal" Bazel paradigm, but is by far the most divergent pattern because of how it would require projects to change their flags (and/or attach flags to platforms).
3. Treat them as features, and just make them no-op `cc_feature`s (rather than `cc_external_feature`, which implies a definition exists externally that shouldn't be clobbered). This is the least elegant, but the most widely compatible with the current ecosystem.

@comius This relates to starlarkification a bit. Is there a long-term vision for how users should control implementation behavior of cc rules for things like this?

Reply:
> Do we really want to add a dedicated attribute for something we've already decided is an anti-pattern though?

I think that the fact that they're an antipattern is precisely why they should get special treatment. It'd be a lot easier to migrate from `cc_target_capability` if they weren't mixed in with other features.

> There's a bunch of fragmentation here. I'm happy with `cc_tool_capability`, with the minor caveat that clearly some tool capabilities can't cleanly be expressed that way today.
> 
> It's starting to feel like we have other paths for these two oddball cases that are target-specific:
> 
> 1. Make the fact that they're features purely an implementation detail, and just expose them as direct bool attributes of a `cc_toolchain` configuration a la `supports_header_parsing`. Users won't be able to express feature relationships with these any more, but I'm not seeing any reason that they should or any instances where Bazel relies on this being possible.
This seems quite reasonable to me. This could possibly be merged with option 2 where if you set `supports_pic = true` in the toolchain config rule, it'd transition to the flag supports_pic = True.

> 2. Maybe they should just be `bool_flag`s, which lets platforms control the behavior directly at the expense of losing any correlation with the toolchain configuration. This seems like the most "normal" Bazel paradigm, but is by far the most divergent pattern because of how it would require projects to change their flags (and/or attach flags to platforms).

I like this option, because it would translate the best to a starlarkified world. It might make them a little more awkward to use for now, but this way people would do selects on them, which matches how we'd expect them to be used when they're fully expressed in starlark.

> 3. Treat them as features, and just make them no-op `cc_feature`s (rather than `cc_external_feature`, which implies a definition exists externally that shouldn't be clobbered). This is the least elegant, but the most widely compatible with the current ecosystem.

I'm not a fan of this option, mainly because anyone using it improperly would give themselves a nightmare for when they do get starlarkified.